apps/web/pages/new-booker/[user]/[type].tsx

```diff
@@ -55,7 +55,15 @@ async function getDynamicGroupPageProps(context: GetServerSidePropsContext) {
     booking = await getBookingByUidOrRescheduleUid(`${rescheduleUid}`);
   }
 
-  await ssr.viewer.public.event.prefetch({ username: user, eventSlug: slug });
+  // We use this to both prefetch the query on the server,
+  // as well as to check if the event exist, so we c an show a 404 otherwise.
+  const eventData = await ssr.viewer.public.event.fetch({ username: user, eventSlug: slug });
+
+  if (!eventData) {
```

Comment:
For my knowledge, then why would someone use prefetch over fetch then ?

Reply:
Prefetch does pretty much te same thing, but doesn't return any data. Using fetch returns the data, but also stores it in the dehydrated state. I guess there's more differences, but in this case I need the data, so prefetch wouldn't work because I can't get the data then. (unless I'd make my own life hard and get the data from the dehydrated state, but that would result in way more and uglier code)